vectordb_bench/backend/clients/redis/redis.py

```diff
@@ -103,20 +102,27 @@ def insert_embeddings(
         """Insert embeddings into the database.
         Should call self.init() first.
         """
+
+        batch_size = 1000 # Adjust this as needed, but don't make too big
```

Comment:
Super minor nit pick, but I think the `i` param is all you need here:

```python
batch_size = 1000 # Adjust this as needed, but don't make too big
try:
    with self.conn.pipeline(transaction=False) as pipe:
        for i, embedding in enumerate(embeddings, start=1):
            embedding = np.array(embedding).astype(np.float32)
            pipe.hset(metadata[i], mapping = {
                "id": str(metadata[i]),
                "metadata": metadata[i], 
                "vector": embedding.tobytes(),
            })
            # Execute the pipe so we don't keep too much in memory at once
            if i % batch_size == 0:
                res = pipe.execute()

        res = pipe.execute()
except Exception as e:
    return 0, e

return i, None

```

Reply:
If I start i =1 then i ends with 5001 instead of 5000. That results in an index out of bounds error by the caller.